docs/contributing/MODEL_MANAGER.md

```diff
@@ -0,0 +1,1214 @@
+exceptions on failure.
+
+***TO DO:*** Investigate why `update_model()` returns an
+`AnyModelConfig` while `add_model()` returns a `ModelConfigBase`.
+
+### rename_model(key, new_name) -> ModelConfigBase:
+
+This is a special case of `update_model()` for the use case of
+changing the model's name. It is broken out because there are cases in
+which the InvokeAI application wants to synchronize the model's name
+with its path in the `models` directory after changing the name, type
+or base. However, when using the ModelRecordService directly, the call
+is equivalent to:
+
+```
+store.rename_model(key, {'name': 'new_name'})
+```
+
+***TO DO:*** Investigate why `rename_model()` is returning a
+`ModelConfigBase` while `update_model()` returns a `AnyModelConfig`.
+
+***
+
+## Let's get loaded, the lowdown on ModelLoadService
+
+The `ModelLoadService` is responsible for loading a named model into
+memory so that it can be used for inference. Despite the fact that it
+does a lot under the covers, it is very straightforward to use.
+
+An application-wide model loader is created at API initialization time
+and stored in
+`ApiDependencies.invoker.services.model_loader`. However, you can
+create alternative instances if you wish.
+
+### Creating a ModelLoadService object
+
+The class is defined in
+`invokeai.app.services.model_loader_service`. It is initialized with
+an InvokeAIAppConfig object, from which it gets configuration
+information such as the user's desired GPU and precision, and with a
+previously-created `ModelRecordServiceBase` object, from which it
+loads the requested model's configuration information.
+
+Here is a typical initialization pattern:
+
+```
+from invokeai.app.services.config import InvokeAIAppConfig
+from invokeai.app.services.model_record_service import ModelRecordServiceBase
+from invokeai.app.services.model_loader_service import ModelLoadService
+
+config = InvokeAIAppConfig.get_config()
+store = ModelRecordServiceBase.open(config)
+loader = ModelLoadService(config, store)
+```
+
+Note that we are relying on the contents of the application
+configuration to choose the implementation of
+`ModelRecordServiceBase`.
+
+### get_model(key, [submodel_type], [context]) -> ModelInfo:
+
+The `get_model()` method, like its similarly-named cousin in
+`ModelRecordService`, receives the unique key that identifies the
+model.  It loads the model into memory, gets the model ready for use,
+and returns a `ModelInfo` object. 
+
+The optional second argument, `subtype` is a `SubModelType` string
+enum, such as "vae". It is mandatory when used with a main model, and
+is used to select which part of the main model to load.
+
+The optional third argument, `invocation_context` can be provided by
+an invocation to trigger model load event reporting. See below for
+details.
+
+The returned `ModelInfo` object shares some fields in common with
+`ModelConfigBase`, but is otherwise a completely different beast:
```

Comment:
Is there a reason for returning these metadata fields other than simply: "they are readily available so might as well return them"?

Reply:
These are the subset of metadata fields needed to load and run the model. 

I'd rather return a `ModelInfo` object that contains a field for the model's `ModelConfigBase` rather than duplicating those fields, but the change would involve multiple detailed changes in Stalker's working model loading code.